internal/contour/holdoff.go

```diff
@@ -63,9 +66,22 @@ func (hn *HoldoffNotifier) OnChange(builder *dag.Builder) {
 	hn.timer = time.AfterFunc(holdoffDelay, func() {
 		hn.mu.Lock()
 		defer hn.mu.Unlock()
-		hn.WithField("last update", time.Since(hn.last)).Info("performing delayed update")
+		hn.WithField("last_update", time.Since(hn.last)).WithField("pending", hn.pending.reset()).Info("performing delayed update")
 		hn.Notifier.OnChange(builder)
 		hn.last = time.Now()
 		hn.Metrics.SetDAGRebuiltMetric(hn.last.Unix())
 	})
 }
+
+// counter holds an atomically incrementing counter.
+type counter uint64
```

Comment:
I considered extracting this into its own package (either `internal/atomic` or `internal/safe` to allow for reuse by `xds.go`.  Something like `pending safe.Uint64`.  I'm happy to do that with this PR or as a secondary one to collapse similar use in other areas.

Reply:
Let's follow Atwood's rule of three here and wait for two more use cases before refactoring.